mne/viz/_3d.py

```diff
@@ -1927,44 +1943,74 @@ def _onclick(event, params):
     plt.ylabel('Activation')
 
-    ctrl_pts, colormap_auto, _, _ = _limits_to_control_points(
-        clim, stc.data, colormap, transparent=True, fmt='matplotlib')
-    if colormap in ('mne', 'mne_analyze'):
-        cmap = colormap_auto
-    else:
-        cmap = _get_pysurfer_cmap(colormap_auto, ctrl_pts)
+    ctrl_pts, colormap, scale_pts, _ = _limits_to_control_points(
+        clim, stc.data, colormap, transparent, fmt='matplotlib')
+    if np.array_equal(ctrl_pts, scale_pts):  # set eq above iff one-sided
+        # there is a bug in nilearn where this messes w/transparency
+        # Need to double the colormap
+        if (ctrl_pts < 0).any():
+            # XXX We should fix this, but it's hard to get nilearn to
+            # use arbitrary bounds :(
+            # Should get them to support non-mirrored colorbars, or
+            # at least a proper `vmin` for one-sided things.
+            # Hopefully this is a sufficiently rare use case!
+            raise RuntimeError('Negative values not supported currently, '
+                               'Consider shifting or flipping the sign of '
+                               'your data for visualization purposes')
+        colormap = plt.get_cmap(colormap)
+        bottom = np.array(colormap(0))
+        locs = np.clip(np.linspace(-ctrl_pts[0] / ctrl_pts[2], 1, 256), 0, 1)
+        colormap = colormap(locs)
+        if transparent:
+            colormap[:, 3] = np.clip(locs * 2, 0, 1)
+            bottom[3] = 0
+        colormap = np.concatenate([np.tile(bottom, (256, 1)), colormap])
+        colormap = colors.ListedColormap(colormap)
+    vmax = ctrl_pts[-1]
 
     # black_bg = True is needed because of some matplotlib
     # peculiarity. See: https://stackoverflow.com/a/34730204
     # Otherwise, event.inaxes does not work for ax_x and ax_z
-    plot_map_callback = partial(
-        plot_func, threshold=ctrl_pts[0], axes=[0.09, 0.55, 0.9, 0.4],
-        resampling_interpolation='nearest', vmax=ctrl_pts[2], figure=fig,
-        colorbar=True, bg_img=bg_img_param, cmap=cmap, black_bg=True)
-
-    params = {'stc': stc, 'ax_time': ax_time,
-              'plot_map_callback': plot_map_callback,
-              'img_idx': img_idx,
-              'fig': fig, 'bg_img': bg_img, 'lx': lx}
-
-    fig_anat = plot_map_callback(
-        stat_map_img=params['img_idx'], title='',
-        cut_coords=cut_coords)
+    axes = fig.add_axes([0.09, 0.55, 0.9, 0.4])
+    plot_kwargs = dict(
+        threshold=None, axes=axes,
+        resampling_interpolation='nearest', vmax=vmax, figure=fig,
+        colorbar=True, bg_img=bg_img_param, cmap=colormap, black_bg=True,
+        symmetric_cbar=True)
+
+    plot_func = partial(plot_func, **plot_kwargs)
+
+    def plot_and_correct(*args, **kwargs):
+        axes.clear()
+        fig.axes[-1].clear()  # cbar
+        with warnings.catch_warnings(record=True):  # nilearn bug; ax recreated
+            warnings.simplefilter('ignore', MatplotlibDeprecationWarning)
```

Comment:
ha! I was trying to get around it with `delaxes` but I guess this is easier ...

Reply:
Yes this is a bug that should be fixed upstream ideally. In the meantime you're right that `delaxes` might be better if it's possible, but this should probably be okay